src/NuGet.Core/NuGet.DependencyResolver.Core/Remote/RemoteWalkContext.cs

```diff
@@ -34,6 +37,7 @@ public RemoteWalkContext(SourceCacheContext cacheContext, ILogger logger)
         public IList<IDependencyProvider> ProjectLibraryProviders { get; }
         public IList<IRemoteDependencyProvider> LocalLibraryProviders { get; }
         public IList<IRemoteDependencyProvider> RemoteLibraryProviders { get; }
+        public PackageNamespacesConfiguration PackageNamespaces { get; set; }
```

Comment:
Can we avoid the setter?

Reply:
`PackageNamespaces` configuration is currently not passed to a constructor while creating `RemoteWalkContext` instance. 
The value is assigned after creating the context object here https://github.com/NuGet/NuGet.Client/pull/4149/files#diff-fdeafb08a0d0ea6b92176fbc5d11e5eb726f77500e857cca8d8a646ff17fd379R118

If we make this a readonly property then we should change, https://github.com/NuGet/NuGet.Client/blob/dev/test/TestUtilities/Test.Utility/DependencyResolver/TestRestoreWalkContext.cs also.

Reply:
I support at least `readonly` property is right thing to do. But not sure how much work it need to be done due to change in `TestRestoreWalkContext.cs`

Reply:
Could we introduce new constructor that accept PackageNamespacesConfiguration so we can remove the setter?

        public RemoteWalkContext(SourceCacheContext cacheContext, PackageNamespacesConfiguration packageNamespacesConfiguration, ILogger logger) : this(cacheContext, logger)
        {
            PackageNamespaces = packageNamespacesConfiguration;
        }

Reply:
I'm fine if we change the constructor.

Reply:
Fixed in the latest commit. Added a new parameter to the existing constructor to pass package namespaces configuration.